folly/Traits.h

```diff
@@ -509,15 +509,8 @@ struct IsRelocatable< std::pair<T, U> >
         IsRelocatable<U>::value> {};
 
 // Is T one of T1, T2, ..., Tn?
-template <class T, class... Ts>
-struct IsOneOf {
-  enum { value = false };
-};
-
-template <class T, class T1, class... Ts>
-struct IsOneOf<T, T1, Ts...> {
-  enum { value = std::is_same<T, T1>::value || IsOneOf<T, Ts...>::value };
-};
+template <typename T, typename ...Ts>
+using IsOneOf = Negation<StrictConjunction<Negation<std::is_same<T, Ts>>...>>;
```

Comment:
Ahh, sorry. I read `StrictConjunction` instead of `StrictDisjunction` and wasn't able to understand how would you like to create `StrictConjunction` without the `Negative`s, my bad.

And yes, there is no `StrictDisjunction` yet. I'll create one and update the pr.